drivers/CpuUid.cpp

```diff
@@ -0,0 +1,70 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2017 ARM Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "hal/cpu_uid_api.h"
+#include "platform/mbed_assert.h"
+#include "platform/CriticalSectionLock.h"
+#include "drivers/CpuUid.h"
+
+#if DEVICE_CPUUID
```

Comment:
I'm concerned about the fact that this may confuse engineers since "ARMv7M Architecture Reference Manual"(other arch versions as well) defines its own CPU identification scheme as defined in "Chapter B4 - The CPUID Scheme". The implementation here seems to be identifying a specific device( For example: Nordic NRF52840 ), is that correct?. The driver here provides identification mechanism for a specific Device/SoC and not CPUID. So naming this class as CpuUid seems concerning, maybe we should name it DeviceUid. Or change the implementation to return CpuID and DeviceID using different routines/functions.

Reply:
Ok didn't know that - I'll check that chapter of the arm rm

Reply:
ok now i understood your question: nope, its not to identify the cpu type ... its about having a serial number to identify an individual IC soldered on a produced pcb

Reply:
Thanks for your reply, in that case shouldn't we be calling this SerialNumber. I'm still not convinced why we are calling this Cpu UID. Changing the name from CpuUid to something more specific( SerialNumber? ) would be really helpful.

Reply:
For the past years I have worked for projects using RFID - in ISO14443 the serial number of a card is called UID, so I found that would be a good fit.
I also thought about calling the driver CpuSerialNumber but that is much longer than CpuUid ;)

Maybe CpuUniqueID ?

Reply:
DeviceUID/TargetUID would be good , @MikeDK what do you think? looking through the implementations for targets, seems like their API name it similar, or from their docs - named as device uid often.